content/technical/open-source-as-a-service.md

```diff
@@ -0,0 +1,155 @@
+Title: Your Service is not Open Source
+Date: 2018-02-06 18:35
+Tags: open source, technical
+Slug: open-source-as-a-service
+Category: Technical
+Summary: Sharing the source code to your SaaS is insufficient. We effectively enable users to contribute to the a running service for it to be Open Source.
+
+Sharing the source code to your SaaS is insufficient to make it
+actually be Open Source. Sounds self-contradictory?
+
+Most services that espouse "Open Source", do so by simply
+[throwing the code over the wall](https://github.com/travis-ci/travis-ci).
+It's better than nothing, but really misses the point that powers
+Open Source: enabling users to make a change to the software
+they're using.
+
+Some other Open Source services, such as
+[GitLab.com](https://about.gitlab.com/community/contribute/development/)
+or [ElasticSearch](https://www.elastic.co/about/free-and-open) do include
+the tools used to operate/deploy their service. Pause for applause 👏👏👏
+
+But thats is also insufficient to actually enable Open Source.
+
+In order for a user of the service to begin contributing, they must
+first deploy and operate their own clone: Something that they explicitly
+chose not to do when they elected to use the software as a service.
+
+Even if the user becomes an operator, and deploys a such clone, it
+is devoid of their own data, devoid of other users, of connections
+to other services, of infrastructure scalability, pipelines, monitoring,
+backups, ... you name it.
+
+In any case, users are certainly not able to iterate on their changes
+to a service against their real world use cases.
+
+What's more, without true Open Source principles, we are missing a
+way to have a community develop with a center of gravity around
+the service itself. This leads to companies
+[forking the service](https://aws.amazon.com/blogs/opensource/stepping-up-for-a-truly-open-source-elasticsearch)
+in a way that typical open source projects with a true community
+are resilient to.
+
+I believe that if we enable contributions to services, we reap
+the advantages of true Open Source. That means enabling users to make a
+change to a running service and experience that change themselves.
+
+You still with me?
+
+Yes, that means opening a pull request against a running service
+and running that change before it's merged.
+
+This is not as insane as it sounds.
+
+It appears that every technique we need to enable such a capability
+is already in use in modern deployment and operations methodology.
+We just have to connect the dots.
+
+
+# Why should I care?
+
+I've been a part of Open Source for over 20 years now. Although I'm
+just one contributor, together we've changed something fundamental
+about the world. Our lives are too short, and software too complex,
+to have one company or one individual invent everything from scratch.
+So we work together across humanity to accomplish what no
+individual team could. This is our legacy, and it has become
+common place.
```

Comment:
I had to shed a tear of joy here :joy: Well written!

Reply:
Thanks!